src/system/sys_profiler.hh

```diff
@@ -214,7 +224,60 @@ inline CONSTEXPR_ unsigned int str_hash(const char * str, unsigned int default_v
  */
 #define CODE_POINT_EXT(tag, subtag) CodePoint(tag, subtag, __FILE__, __func__, __LINE__)
 
+/**
+ * Simple allocator which uses default malloc and free functions. Dields allocated
```

Comment:
Clearly document that this is only internal class. I suggest to move it into separate file and into the namespace 'internal'.